### ISSUE TITLE
Support for libraries without public headers

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle-native#451]
The Gradle C++ library currently requires that a library has at least one public header file to build the cpp-api-headers zip.  Not all libraries may provide a public API in this way.

Some examples include:
- A library that is loaded solely at runtime and implements an interface provided by an external dependency (e.g., a common "plugin" system)
- A library that is loaded at runtime and implements an interface described by something other than header files (e.g., a jar + JNI library)
- A library that shims another library so that it appears to be the other one

**Comment by adammurdoch** [comment]
I don't think we should consider all of these things "libraries". A library is something that provides an API. The last example is a library, because it _does_ have a public API (it "exports" the API of the other library). The other examples are "modules" or "plugins".
